### DISCUSSION TITLE
UI configuration example

### DISCUSSION THREAD

**Discussion by Ince1956** [SignalK/SensESP#792] [Q&A]
Hi
I am looking for some sample code to help me with the UI. I have managed to create a checkbox using CheckboxConfig which appears on the device webpage  however I can't figure out how to store the value and retrieve it when the device reboots.  Also how to access the value when the device is running and connected to the signalK server.

Any help would be greatly appreciated
Thanks
Ian

**Comment by mairas** [comment]
Hi Ian,

The checkbox value should be automatically stored in a file if you set a config_path:

```
auto my_checkbox = new CheckboxConfig(false, "My checkbox", "/my_checkbox");
```

then, you can access the value at runtime by calling `my_checkbox->get_value()`. A CheckboxConfig itself isn't a producer, so you can't connect to it to get its value automatically when it changes. There is a `PersistingObservableValue` that can do that, if you need it. I think it renders bools as checkboxes as well.

**Comment by Ince1956** [comment]
Hi Matti
Thanks for your prompt reply.   How do you save the value and reload after
reboot?
Regards
Ian

**Reply by mairas** [reply]
You probably already found out that this happens automatically. The web server callback calls `Saveable::save()` when new values are received, and likewise, the `Saveable::load()`is called when the object is constructed.

**Comment by Ince1956** [comment]
Thanks, I now have it working.  Is there a way to get the connection status !
Regards
Ian

**Reply by mairas** [reply]
Connect to SystemStatusController. It produces an enum whenever the SK or WiFi connection status changes. (Although, if you'd be only interested in WiFi status, you might prefer registering a native Arduino `Wifi.onEvent` callback.)

**Comment by Ince1956** [comment]
Thanks,  i just wanted to put an indicator on the lcd screen when it has
connected to the signalk server
